src/Foundatio.Repositories.Elasticsearch/Configuration/DailyIndex.cs

```diff
@@ -131,13 +135,19 @@ protected override DateTime GetIndexDate(string index)
         return DateTime.MaxValue;
     }
 
-    private readonly Dictionary<DateTime, object> _ensuredDates = new();
     protected async Task EnsureDateIndexAsync(DateTime utcDate)
     {
         utcDate = utcDate.Date;
         if (_ensuredDates.ContainsKey(utcDate))
             return;
 
+        await using var indexLock = await _ensureIndexLock.AcquireAsync($"Index:{GetVersionedIndex(utcDate)}", TimeSpan.FromMinutes(1)).AnyContext();
```

Comment:
It kind of feels like we should be locking on the index name here, but this allows us to go wide. I'm wondering if we should also change this to a concurrent dictionary.